src/main/scala/GuessWhoGame/Game.scala

```diff
@@ -22,59 +24,71 @@ object Game extends App {
   val char17: Character = Character("roshan", "male", "brown", "blue", wearsGlasses = true, facialHair = false)
   val char18: Character = Character("patricia", "female", "purple", "yellow", wearsGlasses = false, facialHair = false)
   val char19: Character = Character("dave", "male", "none", "yellow", wearsGlasses = false, facialHair = false)
-  val char20: Character = Character("bobbiana", "female", "Brown", "Blue", wearsGlasses = true, facialHair = false)
+  val char20: Character = Character("bobbiana", "female", "brown", "blue", wearsGlasses = true, facialHair = false)
 
+  //would remain the same throughout
   var gameBoard = Board(List(char1, char2, char3, char4, char5, char6, char7, char8, char9, char10, char11, char12, char13, char14, char15, char16, char17, char18, char19, char20))
 
-  gameBoard.printCharacterNames()
-
-
-  //Run example of the game
-  //Call methods
-
-  /** game start - random character selected (private), characterlist printed to screen
-   * yes / no question asked
-   * list filtered based on fulfilment of criteria
-   * list reprinted to screen
-   * loop through all options, filtering and reprinting each time
-   * trigger game end when length of list is 1 ------- could end anytime, defined/earlier in code
-   * */
-
-
-
-//  case class GuessWhoGame.Character(
-//                        name: String,
-//                        gender: String,
-//                        hairColor: String,
-//                        eyeColor: String,
-//                        wearsGlasses: Boolean,
-//                        facialHair: Boolean
-//                      )
-
-
-
-
-
-
-
-
-
+  def startGame(): Unit = {
+    gameBoard.printCharacterNames()
+
+    //Building out game to terminal
+    println("Welcome to the Guess Who game!")
+    println("Try to guess the character by asking some questions...")
+
+    playGame()
+
+    println("\nDo you want to play again? (yes/no)")
+    val playAgain = scala.io.StdIn.readLine().toLowerCase()
+    if (playAgain == "yes" || playAgain == "y") {
+      gameBoard = Board(List(char1, char2, char3, char4, char5, char6, char7, char8, char9, char10, char11, char12, char13, char14, char15, char16, char17, char18, char19, char20))
+      startGame()
+    } else {
+      println("Thanks for playing Guess Who!")
+    }
+  }
+
+  //Game loop
+  def playGame(): Unit = {
+    while (!gameBoard.checkWinCondition) {
```

Comment:
while loops lack the control of recursion, think about how you could refactor this. Tail recursion would be ideal here!

Reply:
Tail recursion has been implemented.